File: 0078-subsets/0078-subsets.py
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        # result = []
        # for i in range(len(nums) + 1) :
        #     result.extend(list(itertools.combinations(nums, i)))
        
        result = list(j for i in range(len(nums) + 1) for j in itertools.combinations(nums, i) )
        
        return result

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int> elements;
        vector<vector<int>> result;
        vector<int>::iterator it = nums.begin();
        
        subsets(nums, elements, result, it); 
        
        return result;
    }
    
    void subsets(vector<int>& nums, vector<int> elements, vector<vector<int>>& result, vector<int>::iterator it) {
        if (it == nums.end()) {
            result.push_back(elements);
            return;
        }
        subsets(nums, elements, result, it+1);
        elements.push_back(*it);
        subsets(nums, elements, result, it+1);
    }
};